lib/mpi/service.rb

```diff
@@ -166,13 +168,21 @@ def create_add_message(user_identity)
     end
 
     def create_profile_message(user_identity, search_type: MPI::Constants::CORRELATION_WITH_RELATIONSHIP_DATA)
-      return message_icn(user_identity, search_type) if user_identity.mhv_icn.present?
-      return message_edipi(user_identity, search_type) if user_identity.edipi.present? && Settings.mvi.edipi_search
+      return message_icn(user_identity, search_type) if mhv_icn_present?(user_identity)
+      return message_edipi(user_identity, search_type) if edipi_present?(user_identity)
       raise Common::Exceptions::ValidationErrors, user_identity unless user_identity.valid?
 
       message_user_attributes(user_identity, search_type)
     end
 
+    def mhv_icn_present?(user_identity)
+      user_identity.respond_to?(:mhv_icn) && user_identity.mhv_icn.present?
```

Comment:
Instead of `respond_to?` for all of these, we can just do the safe access operator: `user_identity&.mhv_icn.present?`

Reply:
Ordinarily yes, but in this case `user_identity` is a struct object and they don't function with the safe operator. It might be a good idea for me to put a comment here clarifying that though so someone doesn't try to refactor it later.

Reply:
Ahhh ok, makes sense